src/Notes/SetupCouponSharing.php

```diff
@@ -0,0 +1,135 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\Notes;
+
+use Automattic\WooCommerce\Admin\Notes\Note as NoteEntry;
+use Automattic\WooCommerce\GoogleListingsAndAds\Ads\AdsAwareTrait;
+use Automattic\WooCommerce\GoogleListingsAndAds\HelperTraits\Utilities;
+use Automattic\WooCommerce\GoogleListingsAndAds\MerchantCenter\MerchantCenterAwareInterface;
+use Automattic\WooCommerce\GoogleListingsAndAds\MerchantCenter\MerchantCenterAwareTrait;
+use Automattic\WooCommerce\GoogleListingsAndAds\MerchantCenter\MerchantStatuses;
+use Automattic\WooCommerce\GoogleListingsAndAds\PluginHelper;
+use Automattic\WooCommerce\GoogleListingsAndAds\Ads\AdsAwareInterface;
+use Automattic\WooCommerce\GoogleListingsAndAds\Value\ChannelVisibility;
+use Automattic\WooCommerce\GoogleListingsAndAds\Coupon\CouponMetaHandler;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class SetupCouponSharing
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\Notes
+ */
+class SetupCouponSharing extends AbstractNote implements MerchantCenterAwareInterface, AdsAwareInterface {
+
+	use AdsAwareTrait;
+	use MerchantCenterAwareTrait;
+	use PluginHelper;
+	use Utilities;
+
+	/** @var MerchantStatuses */
+	protected $merchant_statuses;
+
+	/**
+	 * SetupCouponSharing constructor.
+	 *
+	 * @param MerchantStatuses $merchant_statuses
+	 */
+	public function __construct( MerchantStatuses $merchant_statuses ) {
+		$this->merchant_statuses = $merchant_statuses;
+	}
+
+	/**
+	 * Get the note's unique name.
+	 *
+	 * @return string
+	 */
+	public function get_name(): string {
+		return 'gla-coupon-optin';
+	}
+
+	/**
+	 * Get the note entry.
+	 */
+	public function get_entry(): NoteEntry {
+		$note = new NoteEntry();
+
+		$note->set_title( __( 'Show your store coupons on your Google listings', 'google-listings-and-ads' ) );
+		$note->set_content(
+			__(
+				'Sync your store promotions and coupons directly with Google to showcase on your product listings across the Google Shopping tab. When creating a coupon, you’ll see a Channel Visibility settings box on the right; select "Show coupon on Google" to enable.',
+				'google-listings-and-ads'
+			)
+		);
+		$note->set_content_data( (object) [] );
+		$note->set_type( NoteEntry::E_WC_ADMIN_NOTE_INFORMATIONAL );
+		$note->set_layout( 'plain' );
+		$note->set_image( '' );
+		$note->set_name( $this->get_name() );
+		$note->set_source( $this->get_slug() );
+		$note->add_action(
+			'coupon-views',
+			__( 'View coupons', 'google-listings-and-ads' ),
+			$this->get_coupons_url()
+		);
+		$note->add_action(
+			'coupon-more-info',
+			__( 'Learn more', 'google-listings-and-ads' ),
+			$this->get_documentation_url()
+		);
```

Comment:
This should be changed from two buttons to the [one "Go to coupons" button](https://github.com/woocommerce/google-listings-and-ads/pull/1105#issuecomment-990562495):
![image](https://user-images.githubusercontent.com/228780/145669134-c36151ef-0bee-48ba-bdb1-e415db78f46b.png)

```suggestion
		$note->add_action(
			'coupon-views',
			__( 'Go to coupons', 'google-listings-and-ads' ),
			$this->get_coupons_url()
		);
```